### ISSUE TITLE
mlx_lm.generate.generate_step() omits prompt-prefill tokens from logits-processor history.

### ISSUE THREAD

**Issue by Sayitobar** [ml-explore/mlx-lm#1773]
## Summary

`generate_step()` does not expose the full prompt history to logits processors.

When a multi-token prompt is prefetched, the prefetched tokens are added to the model KV cache but not to the `tokens` value passed to logits processors. As a result, the first processor call sees only the final prompt token instead of the complete prompt.

## Environment

- mlx-lm: 0.31.3
- mlx: 0.32.0
- OS: macOS 15.7.3 on Apple Silicon
- Python: 3.10.18

## Minimal reproduction

The following uses a recorder processor that only records the token history it receives:

```python
import mlx.core as mx
from mlx_lm.generate import generate_step

prompt_ids = mx.array([2, 5, 7, 9])
first_history = []

def record_history(tokens, logits):
    if not first_history:
        first_history.append(tokens.tolist())
    return logits

list(
    generate_step(
        prompt_ids,
        model,  # which is any already-loaded compatible model.
        max_tokens=1,
        logits_processors=[record_history],
    )
)

print(first_history[0])
```

## Actual result

```text
[9]
```

The first processor call receives only the final prompt token.

## Expected result

```text
[2, 5, 7, 9]
```

The processor should receive the full semantic token history, with each prompt token represented exactly once.

## Why this matters

Stateful logits processors such as repetition, presence, and frequency penalties use the supplied token history as their context window.

Because prompt-prefill tokens are missing from that history, these processors cannot apply their configured context window to earlier prompt text. They only see the final prompt token and tokens generated during the current completion.

This is especially problematic for multi-turn generation, where important conversation context may be in the prefetched part of the prompt.

## Implementation detail

In `generate_step()`, the prefill loop calls `_model_call()` directly. The local `tokens` value is updated only inside `_step()`, which is called for the remaining final prompt token and subsequent generated tokens.

This makes the model cache history and the processor-visible token history inconsistent.
